File: FITCC.UI/admin_fitcc/lib/models/komisija.dart
import 'package:admin_fitcc/models/kategorija.dart';
import 'package:admin_fitcc/models/enums/ulogeKomisije.dart';
import 'package:json_annotation/json_annotation.dart';

part 'komisija.g.dart';

@JsonSerializable(explicitToJson: true)
class Komisija {
  int komisijaId;
  int kategorijaId;
  String ime;
  String prezime;
  String email;
  UlogeKomisije role; 
  Kategorija? kategorija; 

  Komisija(
     this.komisijaId,
    this.ime,
    this.prezime ,
    this.email ,
    this.role , 
     this.kategorija, 
     this.kategorijaId, 
  );

  factory Komisija.fromJson(Map<String, dynamic> json) => _$KomisijaFromJson(json);
  Map<String, dynamic> toJson() => _$KomisijaToJson(this);
}

File: FITCCRS2App.Models/Models/Komisija.cs
using FITCCRS2App.Models.Enums;

namespace FITCCRS2App.Models.Models
{
    public class Komisija
	{
        public int KomisijaId { get; set; }
        public string Ime { get; set; } = null!;
        public string Prezime { get; set; } = null!;
        public string Email { get; set; } = null!;

        public UlogaKomisija Role { get; set; } = UlogaKomisija.Clan;

        public int? KategorijaId { get; set; }
        public virtual Kategorija? Kategorija { get; set; }
    }
}